src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs

```diff
@@ -102,6 +104,34 @@ public static Metadata ConvertMetadata(IMetadataReaderSession provider)
 
                         avaresValues.Add(avres);
                     }
+
+                    //add x:Class avalonia resources, they're not in the CompiledAvaloniaXaml.!AvaloniaResources format
+                    //but they are in the NamespaceInfo:
+                    foreach (var nestedType in resType.NestedTypes.Where(t => t.Name.StartsWith("NamespaceInfo:")))
```

Comment:
I don't think that referencing resources this way will work in run time

Reply:
we don't have general support for all types of resources anyway and that's not possible with reflection only with `avares://` way of work by design.

Here the attempt is at least to have some completion for xaml resources like Style and Styles (*.xaml) compiled with x:Class